test/meson.build

```diff
@@ -0,0 +1,38 @@
+gtest_dep = dependency('gtest', main : true, required : false)
+if not gtest_dep.found()
+    gtest_proj = import('cmake').subproject('googletest', required: false)
+    if gtest_proj.found()
+        gtest_dep = declare_dependency(
+            dependencies: [
+                dependency('threads'),
+                gtest_proj.dependency('gtest'),
+                gtest_proj.dependency('gtest_main'),
+            ]
+        )
+    else
+        assert(
+            not get_option('tests').enabled(),
+            'Googletest is required if tests are enabled'
+            )
+    endif
+endif
+
+testSourceFiles = [
```

Comment:
lowerCamelCase is not recommended by the meson, use snake case: Ref https://mesonbuild.com/Style-guide.html#naming-variable

the comment applicable to all places.